memiavl/db.go

```diff
@@ -203,9 +203,56 @@ func Load(dir string, opts Options) (*DB, error) {
 		}
 	}
 
+	if err := removeAllTmpDirs(db.dir); err != nil {
+		db.logger.Error("failed to remove tmp directories", "err", err)
+	}
+
 	return db, nil
 }
 
+func removeAllTmpDirs(rootDir string) error {
+	const maxRemovals = 1000
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	var errors []error
+	sem := make(chan struct{}, maxRemovals)
+
+	err := filepath.Walk(rootDir, func(path string, info os.FileInfo, err error) error {
```

Comment:
`os.ReadDir()` is better, `Walk` will traverse recursively into the directories.